calico/felix/fetcd.py

```diff
@@ -207,6 +213,39 @@ def _periodically_report_status(self):
                 sleep_time = interval + jitter
                 gevent.sleep(sleep_time)
 
+    @logging_exceptions
+    def _periodically_usage_count(self):
+        """
+        Greenlet: periodically report the cluster existence to
+        projectcalico.org.
+
+        :return: Does not return, unless reporting disabled.
+        """
+        _log.info("Started usage count thread. Waiting for config.")
+        self._watcher.configured.wait()
+        interval = 86400 # Once every 24 hours
+        gevent.sleep(15) # Sleep for 15 seconds to give estimated_host_count time
+        _log.debug("Usage count interval: %s", interval)
+
+        while True:
+            try:
+                config = self._config
+                calico_version = str(pkg_resources.require("calico")[0].version)
+                _log.info("calico_version = %s", calico_version)
+                felix_count = self._watcher.estimated_host_count()
+                _log.info("felix_count = %s", felix_count)
+                usage_count(calico_version, config.HOSTNAME, config.CLUSTER_GUID, felix_count, config.CLUSTER_TYPE)
+            except:
+                _log.warning("Error when usage_count."
+                             "retrying after %s seconds.", RETRY_DELAY)
+                self.reconnect()
```

Comment:
I don't think you should be calling reconnect here.  That will reconnect to etcd, but we didn't fail to talk to etcd (and reconnect belongs to the main thread, not this one).